### ISSUE TITLE
Cannot find package '@actions/core'

### ISSUE THREAD

**Issue by gadicc** [GaelGirodon/ci-badges-action#4]
Hey, me again :grin: 

I got this with v1.6.0:

```
Run gaelgirodon/ci-badges-action@1.6.0
node:internal/modules/esm/resolve:854
  throw new ERR_MODULE_NOT_FOUND(packageName, fileURLToPath(base), null);
        ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@actions/core' imported from /home/runner/work/_actions/gaelgirodon/ci-badges-action/1.6.0/src/index.js
    at packageResolve (node:internal/modules/esm/resolve:854:9)
    at moduleResolve (node:internal/modules/esm/resolve:92[7](https://github.com/gadicc/discourse2/actions/runs/11348587963/job/31562704064#step:7:8):18)
    at defaultResolve (node:internal/modules/esm/resolve:1157:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:3[8](https://github.com/gadicc/discourse2/actions/runs/11348587963/job/31562704064#step:7:9)3:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:352:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:227:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:87:3[9](https://github.com/gadicc/discourse2/actions/runs/11348587963/job/31562704064#step:7:10))
    at link (node:internal/modules/esm/module_job:86:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v20.13.1
```

I had the same issue when I ran from the sha before, but thought it was because I was using an unreleased version.  That's why I had the extra step:

```yaml
      - name: Prepare packages for next step (remove after non-sha release)
        run: cd /home/runner/work/_actions/gaelgirodon/ci-badges-action && npm i -D @actions/core @actions/github @actions/glob
```

I don't know that much about GitHub actions, so not sure if I'm doing something wrong or where the issue is.

Thanks!

**Comment by GaelGirodon** [comment]
Hey!

**TLDR;** you should use the `v1` tag (or `v1.X.Y` tags if you want to target a specific version) → `uses: gaelgirodon/ci-badges-action@v1` (the `v` is important)

-----

GitHub Actions doesn't install NPM dependencies during workflow execution, [they should be embedded in the action repository](https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-javascript-action#commit-tag-and-push-your-action-to-github), so I maintain 2 separate branches:

- `main` with the action source code only, detailed commit history, and `X.Y.Z` version tags
- `release` with the action source code **and a dependency-free bundle** (built using _ncc_ to bundle the action source code and its dependencies in a single `index.js` file) and `vX.Y.Z` version tags that should be used in workflows.

The `v1` tag is updated to the latest `v1.X.Y` tag on each release, it always targets the latest release so users don't have to update the action version in their workflows every time I push a minor release.

I could have pushed the bundle to the same branch, but I don't like the idea of pushing "compiled" code and dependencies to my development branch, it goes a bit against basic Git best practices 😅

**Comment by gadicc** [comment]
Ahhhh got it!  Thanks so much for the detailed answer 🙏  And quick response.

Had definitely intended to put the `v` there 😅😅  And assumed all was ok when the package downloaded correctly.  Your message explains everything.  Thanks so much again for the package and your time!
